### ISSUE TITLE
Verify domain before requesting certificate

### ISSUE THREAD

**Issue by leifnel** [traefik/traefik#7356]
### Do you want to request a *feature* or report a *bug*?

Feature

### What did you expect to see?

Traefik requests certificate generation and renewals without checking if the domain resolves to itself.
This can cause letsencrypt to block the server if traefik contains domains not yet or not anymore hosted on this server.

Especially when moving an existing domain, that could cause a chicken-and-egg problem as the domain shouldn't be moved to the new server before the keys work, and keys can't be requested before the domain works.

**Comment by SantoDE** [comment]
Hello @leifnel,

thanks for your interest in the project. The idea looks interesting, I have a question tough. Is the problem you're facing something you really did encounter, or is it something you can think about? If yes: In which situation did you encounter it?

Thanks a lot!

**Comment by nekuz0r** [comment]
It's a situation I do encounter often, we are hosting websites for numerous of our clients, but we don't always control the DNS.
Sometimes the client change the DNS without telling us, or we deploy the website before the DNS configuration or propagation is completely done, leading to the situation described by @leifnel.

**Comment by cosmosgenius** [comment]
This happened to me when i got hit by the late limit because i forgot to update my dns settings. Traefik was reloading because i was working with watch mode, and it kept requesting cert which kept failing.

**Comment by leifnel** [comment]
I ran into this problem again.
I copied the setup from one server to another, and forgot to remove the manually set up domains.
Traefik repeatedly tried to get certs for domains hosted elsewhere, and now I'm blocked.

I also noticed that traefik does not see the errormessage telling there were too many attempts, but repeatedly tries again and again every 30 seconds or so. I'm afraid letsencrypt someday gets annoyed over this, and extends the cool-off period if we keep hammering it like this. traefik should wait at least 5 minutes before trying again in case it sees this error message.
